src/searchUtils.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import type { SearchResult } from '@jsforce/jsforce-node';
+import { HumanSearchReporter } from './reporters/search/humanSearchReporter.js';
+import { JsonSearchReporter } from './reporters/search/reporter.js';
+import { CsvSearchReporter } from './reporters/search/csvSearchReporter.js';
+
+export const displaySearchResults = (queryResult: SearchResult, resultFormat: 'human' | 'json' | 'csv'): void => {
```

Comment:
if you wanted to do this without classes, this would be a function that decides which reporter function(ux, results) to call.  

then you wouldn't need all the `new` and `display()` etc both here and in the tests